### PR TITLE
Remove uneeded calls on FS

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Remove unneeded calls on FS

## Motivation and Context
Remove unneeded calls on FS for optimizations

## How to test this PR?
There are no special instructions since this is just a code-review change.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]
LGTM

File: cmd/fs-v1-helpers.go
Comment (ebozduman) [inline cmd/fs-v1-helpers.go:232]: `getObjectInfo` function calls `fsIsDir`.

**Review by ebozduman — DISMISSED** [review]

File: cmd/fs-v1-helpers.go
Comment (harshavardhana) [reply cmd/fs-v1-helpers.go:232]: It doesn't @ebozduman

Comment (harshavardhana) [reply cmd/fs-v1-helpers.go:232]: That has been removed, otherwise this PR won't build.

File: cmd/fs-v1-helpers.go
Comment (nitisht) [inline cmd/fs-v1-helpers.go:425]: are we now ignoring the error if sourcePath is not found?

File: cmd/fs-v1-helpers.go
Comment (Praveenrajmani) [inline cmd/fs-v1-helpers.go:409]: We could remove the `checkPathLength` calls in fsRenameFile, we are already doing this in renameAll()

File: cmd/fs-v1-helpers.go
Comment (harshavardhana) [reply cmd/fs-v1-helpers.go:425]: No renameAll returns same error.

File: cmd/fs-v1-helpers.go
Comment (harshavardhana) [reply cmd/fs-v1-helpers.go:409]: This PR is not going to do that @Praveenrajmani - the intention was to avoid Stat() calls in this PR

**Review by nitisht — APPROVED** [review]

**Comment by nitisht** [comment]
waiting for mint run before merge

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#67cd74c5-f5f1-4c77-a747-b499d4c684b2) |
| mint-large-bucket.sh | :x: [more...](#dfa20902-5bf1-4210-b258-44867a9b43a4) |
| mint-dist-xl.sh | :x: [more...](#929c4781-53e2-4a66-904f-a3fe3ad6ad93) |
<a name="67cd74c5-f5f1-4c77-a747-b499d4c684b2"></a>
### 7967-7786bfe/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32191
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 79ac799286e8:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 13 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ynzdggkstff4md",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 12545,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32191/aws-sdk-go-test-ynzdggkstff4md: dial tcp 72.28.97.57:32191: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="929c4781-53e2-4a66-904f-a3fe3ad6ad93"></a>
### 7967-7786bfe/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30265
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 79afc0cf2942:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 15 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 51 seconds
(8/14) Running minio-dotnet tests ... done in 40 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 13 seconds
(10/14) Running minio-java tests ... FAILED in 46 seconds
{
  "name": "minio-java",
  "function": "removeIncompleteUpload(String bucketName, String objectName)",
  "duration": 243,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-1h2lbmr, objectName=minio-java-test-2r96g17, resource=/minio-java-test-1h2lbmr/minio-java-test-2r96g17, requestId=15B45098764BDBA1, hostId=0933743f-3165-4b0a-adb3-23e47ff447f3)\nrequest={method=DELETE, url=http://72.28.97.61:30265/minio-java-test-1h2lbmr/minio-java-test-2r96g17?uploadId=22dde76a-e9ea-498b-9b81-f31f9f4caf59, headers=Host: 72.28.97.61:30265\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190724T101946Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190724/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 363\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-07-24T10-01-24Z\nVary: Origin\nX-Amz-Request-Id: 15B45098764BDBA1\nX-Xss-Protection: 1; mode=block\nDate: Wed, 24 Jul 2019 10:19:46 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.removeIncompleteUploads_test(FunctionalTest.java:1719), FunctionalTest.runTests(FunctionalTest.java:2992), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```
<a name="dfa20902-5bf1-4210-b258-44867a9b43a4"></a>
### 7967-7786bfe/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30277
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c5801ce84971:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 4 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 13 seconds
(5/14) Running awscli tests ... done in 1 minutes and 29 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 1 minutes and 16 seconds
{
  "name": "mc",
  "duration": "71633",
  "function": "test_put_object_with_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-6091 --quiet --no-color cp --storage-class REDUCED_REDUNDANCY /mint/data/datafile-1-MB myminio/mc-test-bucket-9610/mc-test-object-19951  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-9610/mc-test-object-19951`\nmc: <ERROR> Failed to copy `/mint/data/datafile-1-MB`. Please reduce your request\nmc: <ERROR> Session safely terminated. To resume session `mc session resume KcRqaTnB`"
}

Executed 6 out of 14 tests successfully.
```

**Pull request merged**
